### PR TITLE
feature: Allow customizing trial component display names for pipeline launched jobs

### PR DESCRIPTION
**Pull request by navaj0**
…aunched jobs

*Issue #, if available:*

*Description of changes:* Allow users to customize trial component display names for pipeline launched jobs

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/steps.py [hunk 1]
```diff
@@ -660,7 +669,17 @@ def arguments(self) -> RequestType:
...
+
+        # only keep the trial component name
+        if request_dict.get("ExperimentConfig", {}).get("TrialComponentDisplayName"):
+            request_dict["ExperimentConfig"] = {
+                "TrialComponentDisplayName": request_dict["ExperimentConfig"][
+                    "TrialComponentDisplayName"
+                ]
+            }
+        else:
+            request_dict.pop("ExperimentConfig", None)
```
Comment (shreyapandit) [inline src/sagemaker/workflow/steps.py:681]: This is an entirely repeated block of code, can we refactor this?

File: src/sagemaker/workflow/steps.py [hunk 2]
```diff
@@ -808,7 +827,17 @@ def arguments(self) -> RequestType:
...
+
```
Comment (shreyapandit) [inline src/sagemaker/workflow/steps.py:840]: same here, maybe worth having a outside utils entity?

File: tests/data/_repack_model.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Repack model script for training jobs to inject entry points"""
+from __future__ import absolute_import
+
+import argparse
+import os
+import shutil
+import tarfile
```
Comment (shreyapandit) [inline tests/data/_repack_model.py:19]: Why aren't we using existing repack from the SDK?

Comment (navaj0) [reply tests/data/_repack_model.py]: Thanks for catching this. It is actually generated by the tests but not cleaned up somehow.

File: src/sagemaker/workflow/steps.py (hunk 2 above)
Comment (navaj0) [reply src/sagemaker/workflow/steps.py]: Yes. Refactored it out.

File: src/sagemaker/transformer.py [hunk 4]
```diff
@@ -194,6 +197,11 @@ def transform(
                 (default: ``True``).
             logs (bool): Whether to show the logs produced by the job.
                 Only meaningful when wait is ``True`` (default: ``True``).
+            kms_key (str): The ARN of the KMS key that is used to encrypt the
```
Comment (qidewenwhen) [inline src/sagemaker/transformer.py:200]: Don't see the kms_key in the .transform() signature. Why we add it here in doc string?

Comment (navaj0) [reply src/sagemaker/transformer.py]: Copy/Paste error. Removed it.

**Review by qidewenwhen — APPROVED** [review]

**Pull request merged**